### DISCUSSION TITLE
optee recipe to build from source

### DISCUSSION THREAD

**Discussion by akhilveeraghanta** [OE4T/meta-tegra#1100] [Q&A] [answered]
The previous release of jetpack with Trusty had a recipe that could build the os and example apps from source. It was super convenient to be able to modify the examples apps and also add new programs.

In the newer releases with optee, it looks like the recipe just grabs a prebuilt version of the os/apps with no recipes setup to build from source.

I'm wondering what the recommended approach is to modifying/adding trusted apps w/ optee on L4T 35.1?

I imagine I'd have to write recipes similar to trusty-l4t for optee to build from source, but I just wanted to double check if there is an existing way

Thanks!

**Comment by madisongh** [comment, marked as answer]
We already have a PR pending for this - see #1049. It's still pending because we're seeing issues with system hangs running the OP-TEE test suite when we build from source using the toolchains we have in the kirkstone and master branches.

**Reply by madisongh** [reply]
We now have recipes in master and kirkstone for building OP-TEE from sources, and they are used by default (rather than using the pre-built NVIDIA binaries).
